File: Semestrul 3/MAP/lab 10 MAP/model/ContainerSuperclass.cs
using System.Collections;

namespace seminar10.model;

public abstract class ContainerSuperclass : IContainer
{
    private readonly ArrayList _tasks = new();

    public abstract Task Remove();

    public abstract void Add(Task task);

    public int Size()
    {
        return _tasks.Count;
    }

    public bool IsEmpty()
    {
        return _tasks.Count == 0;
    }
}

File: Semestrul 3/MAP/lab 2 MAP/model/ContainerSuperclass.java
package sem1_2.model;

import java.util.ArrayList;
import java.util.List;

public abstract class ContainerSuperclass implements Container {
    List<Task> tasks = new ArrayList<>();

    @Override
    public abstract Task remove();

    @Override
    public abstract void add(Task task);

    @Override
    public int size() {
        return tasks.size();
    }

    @Override
    public boolean isEmpty() {
        return tasks.isEmpty();
    }
}
